openedx/edx-enterprise-data
# [ENT-979] Use Enterprise Data API rather than Vertica for reporting.

*Testing Instructions*:

1. As usual, set up all the proper Enterprise-related config in your LMS, e.g. an Enterprise Customer, a reporting config, etc.. Make sure the reporting config is for CSV `progress_v2` reports.
1. Clone this repo at this branch.
1. Set up a local version of https://github.com/edx/edx-analytics-data-api.
1.1. Clone the repo and make a virtual environment for it (python 2.7).
1.2. `make develop`
1.3. `pip install -Ue /path/to/edx-enterprise-data/on/your/machine` (the `make develop` command will have already installed a different version than this branch's, so we need to override it with this)
1.4. `./manage.py migrate --noinput`
1.5. `./manage.py migrate --noinput --database=analytics`
1.6. Temporarily change `analyticsdataserver.settings.local.LMS_BASE_URL` to have the proper port of your LMS, if necessary (e.g. change from 8000 to 18000 for the docker devstack).
1.7. Temporarily change `/path/to/edx-enterprise-data/enterprise_data/fixtures/enterprise_enrollment.json` to have the `enterprise_id` for the last 2 objects match your own Enterprise Customer's UUID.
1.8. `python manage.py loaddata /path/to/edx-enterprise-data/enterprise_data/fixtures/enterprise_enrollment.json --database analytics`
1.8. `./manage.py runserver` (if your LMS is on port 8000, you need to specify the port here).
2. As in the last PR, modify `enterprise_reporting.send_enterprise_reports.should_deliver_report` to `return True` by default.
3. Put `from pdb import set_trace;set_trace()` below `data_report_zipped = super().send(files)` inside of the delivery methods so you stop short of actually delivering anything.
4. Run `LMS_OAUTH_KEY=discovery-key LMS_OAUTH_SECRET=discovery-secret LMS_ROOT_URL=http://localhost:18000 LMS_OAUTH_HOST=http://localhost:18000 LMS_FERNET_KEY='DUMMY KEY CHANGE BEFORE GOING TO PRODUCTION' SEND_EMAIL_FROM='uman@mslm.me' ANALYTICS_API_URL=http://localhost:8000 python enterprise_reporting/send_enterprise_reports.py`
5. Check the `/tmp` directory for the zip file, open it up with the appropriate password, and ensure the columns and rows look good.
